### PR TITLE
Allow markdown in caption

### PR DESCRIPTION
**Pull request by amykapernick**
Added parsing of caption, to allow for adding markdown in caption, eg. links, bold, italic, etc.

### CODE REVIEW

**Comment by amykapernick** [comment]
Happy to discuss alternative ways to do this, great plugin, and the best that I've been able to find!

**Comment by florianeckerstorfer** [comment]
Thanks for the contribution, using Markdown in captions was something that I wanted to implement for a while now.

1. At the moment `remark` and `remark-html` are defined as dev dependencies (they are used for the tests), not as dependencies. Before I can merge this PR you would need to add them as dependencies.
2. It would be great if you could add tests for this feature as well.
3. After looking at your PR I was thinking if we should make this configurable. I think we have a few possible options (as in plugin options) here: have an option `enableMarkdownCaptions`, an option to pass an array of remark plugins or instead of including the Markdown processing in this plugin, we provide an option `processCaption`, which is a function `(caption: string) => string`. The user of this plugin could then define their own Remark plugin chain inside a function and pass it to `remark-images`. If the user does not provide a function the caption is not processed.

What do you think? If we would add the `processCaption` option we would not need to add `remark` and `remark-html` as dependencies.

**Comment by florianeckerstorfer** [comment]
And also thank you for the feedback about this plugin. I appreciate it very much.

File: README.md [hunk 1]
```diff
@@ -54,19 +54,35 @@ const processor = remark().use([[images, options]]);
...
+| `srcDir`              | `.`                    | Directory where to look for images                                                        |
+| `targetDir`           | `.`                    | Directory where to save generated images                                                  |
 | `loadingPolicy`       | `"eager"`              | Sets the `loading` attribute on `<img>`, `"lazy"` to load images when they become visible |
-| `elasticContainer`    | `true`                 | Insert elastic container to avoid layout jumps when the image loads |
-| `blurredBackground`   | `true`                 | Add a blurred background while the image is loading |
+| `elasticContainer`    | `true`                 | Insert elastic container to avoid layout jumps when the image loads                       |
+| `blurredBackground`   | `true`                 | Add a blurred background while the image is loading                                       |
+| `parseCaption`        | `(caption) => caption` | Define a function to process image caption, eg. convert markdown to HTML                  |
```
Comment (florianeckerstorfer) [inline README.md:70]: I think you forgot to update here with the new name of the option:

```suggestion
| `processOption`        | `(caption) => caption` | Define a function to process image caption, eg. convert markdown to HTML                  |
```

**Review by florianeckerstorfer — CHANGES_REQUESTED** [review]
Awesome, could you update the option name in the README and then this is good to merge.

**Review by florianeckerstorfer — APPROVED** [review]

**Comment by florianeckerstorfer** [comment]
@amykapernick Thanks for the update, I will release a new version ASAP

**Pull request merged**